### DISCUSSION TITLE
Notifying Subscription state Change

### DISCUSSION THREAD

**Discussion by AkbarMirkar** [eclipse-iceoryx/iceoryx#1544] [Q&A]
Hello,

We are integrating the Iceoryx with ara::com. I have a question regarding subscription state.
In between working communication If Publisher goes down, is there any mechanism present so that the subscriber will receive the notification regarding the subscription state change?

Thanks!

**Comment by elfenpiff** [comment]
@AkbarMirkar you can implement exactly this with our service discovery which informs you whenever a service is offered or has stopped. Take a look at our `icediscovery` example: https://github.com/eclipse-iceoryx/iceoryx/tree/master/iceoryx_examples/icediscovery
